### ISSUE TITLE
CREATE USER allows username over 32 chars, then SELECT throws error

### ISSUE THREAD

**Issue by christopherreay** [dolthub/dolt#8120]
Hi first issue, thanks for the great product.

so here is reproducible in `dolt sql`:

```sql
create user "abcdefghijklmnopqrstuvwxyz0123456789"@'localhost' identified by 'abc123';
Query OK, 0 rows affected (0.02 sec)
select User from mysql.user ;
string 'abcdefghijklmnopqrstuvwxyz0123456789' is too large for column 'char(32) CHARACTER SET utf8mb3 COLLATE utf8mb3_bin'
```

the actual user works perfectly well in terms of connecting/authenticating and running queries. Personally I would prefer to be able to have usernames longer than 32 chars, but either way is fine.

**Comment by fulghum** [comment]
Hi @christopherreay, thanks for reporting this issue and providing the repro script. 🙏 

Since MySQL limits usernames to 32 characters, we'll go ahead and get this patched up so that Dolt also validates the user name length on the call to `CREATE USER`. We're generally open to extending MySQL and having larger limits if customers do need them, so let us know if this limit does cause a problem, otherwise we'll get Dolt patched up to match MySQL's behavior here.

**Comment by fulghum** [comment]
I've got this one queued up for the next release. I'm going to release it as a minor version bump, just in case anyone is relying on the old (unvalidated) behavior. We're waiting for one more change to land tomorrow, but if you need this out sooner than that, we're happy to run a separate release for you.

**Comment by fulghum** [comment]
This fix has been released in [Dolt version 1.42.0](https://github.com/dolthub/dolt/releases/tag/v1.42.0)

Thanks for taking the time to report the issue so we could fix it. 🙏  Let us know if there's anything else we can do to help you build with Dolt!
